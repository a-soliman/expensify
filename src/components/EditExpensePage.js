import React from 'react';

const EditExpensePage = () => (
    <div>
        This is from Edit page.
    </div>
); 

export default EditExpensePage;