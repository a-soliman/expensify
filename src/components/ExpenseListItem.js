import React from 'react';

const ExpenseListItem = ({description, amount, createdAt}) => (
    <div>
        <h1>{description}</h1>
        <p>{amount}</p>
        <p>{createdAt}</p>
    </div>
);

export default ExpenseListItem;
