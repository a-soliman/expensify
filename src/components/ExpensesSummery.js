import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import numeral from 'numeral';
import getVisibleExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummery = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal /100).format('$0,0.00');
    
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viwing <span>{expenseCount}</span> {expenseWord}, with a taotal of <span>{formattedExpensesTotal}</span></h1>
                <div className="page-header__action">
                    <Link to="/create" className="button">Add Expense</Link>
                </div>
            </div>
        </div>
    );
};


const mapStateToProps = (state) => {
    const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);
    
    return  {
        expenseCount: visibleExpenses.length,
        expensesTotal: getExpensesTotal(visibleExpenses)
    }
    
};

export default connect(mapStateToProps)(ExpensesSummery);