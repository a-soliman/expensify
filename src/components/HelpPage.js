import React from 'react';

const HelpPage = () => (
    <div>
        HELP Page!
    </div>
);

export default HelpPage;