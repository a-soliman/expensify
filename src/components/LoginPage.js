import React from 'react';

export class LoginPage extends React.Component {
    login = () => {
        console.log('Loggin in...')

        // only if success
        
    };

    render() {
        return (
            <div>
                <h1>Login Page</h1>
                <button onClick={this.login} >Login</button>
            </div>
        );
    }
}