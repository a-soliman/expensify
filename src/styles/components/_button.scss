
button {
    font-size: $m-size;
    color: #fff;
    border: none;
    padding: $m-size;
    font-weight: 300;
    font-size: $font-size-large;
}

.button {
    @extend button;
    background-color: $light-blue;
}