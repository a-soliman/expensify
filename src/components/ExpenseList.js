import React from 'react';
import { connect } from 'react-redux';

import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div className="content-container">
        <div className="list-header">
            <div className="mobile-only">Expenses</div>
            <div className="desktop-only">Expense</div>
            <div className="desktop-only">Amount</div>    
        </div>
        <div className="list-body">
            {
                props.expenses.length === 0 ? (
                    <div className="list-item--message">
                        <span>No Expenses!</span>
                    </div>
                ) : (
                    props.expenses.map(expense => {
                        return <ExpenseListItem key={expense.id} {...expense} />
                    })
                )
            }
        </div>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);

