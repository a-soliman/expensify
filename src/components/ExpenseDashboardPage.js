import React from 'react';
import ExpensesSummery from './ExpensesSummery';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const ExpenseDashboardPage = () => (
    <div>
        <ExpensesSummery />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;