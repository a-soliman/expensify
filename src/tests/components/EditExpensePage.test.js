import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensePage';
import expenses from '../fixtures/expenses';

describe('EditExpensePage Component', () => {
    const expense = expenses[2];
    let editExpense, removeExpense, history, wrapper

    beforeEach(() => {
        editExpense = jest.fn();
        removeExpense = jest.fn();
        history = { push: jest.fn() };
        wrapper = shallow(
            <EditExpensePage 
                expense={expense} 
                editExpense={editExpense} 
                removeExpense={removeExpense} 
                history={history} />
        );
    });

    test('Should render EditExpensePage component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Should handle onSubmit', () => {
        const updates = {
            ... expense,
            description: 'Updated Expense',
            note: 'updated expense'
        };
        wrapper.find('ExpenseForm').prop('onSubmit')(updates);
        expect(history.push).toHaveBeenLastCalledWith('/');
        expect(editExpense).toHaveBeenLastCalledWith(expense.id, updates);
    });

    test('Should handle onRemove', () => {
        wrapper.find('button').simulate('click');
        expect(history.push).toHaveBeenLastCalledWith('/');
        expect(removeExpense).toHaveBeenLastCalledWith({id: expense.id});
    });
});