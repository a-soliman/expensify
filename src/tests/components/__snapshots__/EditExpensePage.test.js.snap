// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditExpensePage Component Should render EditExpensePage component correctly 1`] = `
<div>
  Editing Expense with id of 
  3
  !
  <ExpenseForm
    expense={
      Object {
        "amount": 4500,
        "createdAt": 345600000,
        "description": "Credit Card",
        "id": "3",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
